README.md

```diff
@@ -9,270 +9,314 @@ Benefits
 
-```bash
-git submodule add https://github.com/package_name.git packages/package_name
-```
+   ```bash
+   git submodule add https://github.com/nomad-coe/electronic-parsers.git packages/nomad-parser-plugins-electronic
 
-For example:
+   git submodule add https://github.com/FAIRmat-NFDI/nomad-measurements.git packages/nomad-measurements
+   ```
 
-```bash
-git submodule add https://github.com/nomad-coe/atomistic-parsers.git packages/nomad-parser-plugins-atomistic
-```
 
-Repeat for all local dev packages (e.g., nomad-parser-plugins-electronic, nomad-parser-plugins-atomistic, etc.).
+3. Modify `pyproject.toml`
 
-2. Modify pyproject.toml
+   To ensure `uv` recognizes the 
+   local plugins, we need to make some modifications in the `pyproject.toml`.
+   These include adding the plugin package to `[project.dependencies]` and
+   `[tool.uv.sources]` tables.
+   For the packages listed under `[tool.uv.sources]`, `uv` uses the local code
+   directory made available under `packages/` with the previous
+   step. 
 
-Ensure uv recognizes the local packages by modifying the `pyproject.toml`:
+   Some of the plugins are already listed under 
+   `[project.dependencies]`. If you want to develop one of them, you
+   have to add them under `[tool.uv.sources]`. We do this for `nomad-parser-plugins-electronics`.
 
-```toml
-[tool.uv.workspace]
-members = ["packages/*"]
+   ```toml
+   [tool.uv.sources]
+   ...
+   nomad-parser-plugins-electronic = { workspace = true }
+   ```
 
-[tool.uv.sources]
-nomad-lab = { workspace = true }
-nomad-parser-plugins-electronic = { workspace = true }
-nomad-parser-plugins-atomistic = { workspace = true }
-nomad-parser-plugins-workflow = { workspace = true }
-nomad-parser-plugins-database = { workspace = true }
-...
-```
+   If you're developing a plugin **not** listed under `[project.dependencies]`, you 
+   must first add it as a dependency. After adding the dependencies, update the 
+   `[tool.uv.sources]` section in your `pyproject.toml` file to reflect the new 
+   plugins. You can either add it manually abd run `uv sync`:
 
-> [!NOTE]
-> If you're developing a plugin not listed under [project.dependencies], you must first add it as a dependency. You can do this with uv:
+   ```toml
+   [project]
+   dependencies = [
+   ...
+   "nomad-measurements",
+   ]
+   
+   [tool.uv.sources]
+   ...
+   nomad-measurements = { workspace = true }
+   ```
 
-```bash
-uv add nomad-measurements
-uv add https://github.com/FAIRmat-NFDI/nomad-parser-vasp --branch develop
-```
+   Or, you can use `uv add` which adds the dependency and the source in `pyproject.toml` 
+   and sets up the environment:
 
-After adding the dependencies, update the [tool.uv.sources] section in your pyproject.toml file to reflect the new plugins:
+   ```bash
+   uv add nomad-measurements
+   ```
 
-```toml
-dependencies = [
-  ...
-  "nomad-measurements",
-  "nomad-parser-vasp",
-]
+   > [!NOTE]
```

Comment:
I think the indentation messed up the note here, it doesn't get rendered correctly anymore